lib/SILGen/SILGenExpr.cpp

```diff
@@ -3805,10 +3817,13 @@ SILGenModule::emitKeyPathComponentForDecl(SILLocation loc,
     if (!var->getDeclContext()->isTypeContext()) {
       componentTy = storage->getStorageInterfaceType()->getCanonicalType();
     } else {
-      componentTy = baseTy->getTypeOfMember(SwiftModule, var)
-        ->getReferenceStorageReferent()
-        ->getCanonicalType(genericEnv ? genericEnv->getGenericSignature()
-                                      : nullptr);
+      componentTy =
+        GenericEnvironment::mapTypeIntoContext(genericEnv, baseTy)
```

Comment:
Do you really need to map it into context and map it out again here? Maybe the callers of getTypeOfMember() should do the superclass dance and getTypeOfMember() should always take an exact type.